core/src/util/extrude.cpp

```diff
@@ -1,61 +1,38 @@
 #include "util/extrude.h"
 
-#include "util/floatFormatter.h"
-
+#include "util/yamlUtil.h"
 #include <cmath>
-#include <cstdlib>
 
 namespace Tangram {
 
-Extrude parseExtrudeString(const std::string& _str) {
-
+Extrude parseExtrudeNode(const YAML::Node& node) {
     // Values specified from the stylesheet are assumed to be meters with no unit suffix
     float first = 0, second = 0;
 
-    if (_str == "true") {
-        // "true" means use default properties for both heights, we indicate this with NANs
-        return Extrude(NAN, NAN);
-    }
-
-    if (_str == "false") {
+    bool extrudeBoolean = false;
+    if (YamlUtil::getBool(node, extrudeBoolean)) {
+        if (extrudeBoolean) {
+            // "true" means use default properties for both heights, we indicate this with NANs
+            return Extrude(NAN, NAN);
+        }
         // "false" means perform no extrusion
         return Extrude(0, 0);
     }
 
-    // Parse the first of two possible numbers
-    const char* str = _str.c_str();
-
-    // Get a float if possible & advance pos to the end of the number
-    const int length = _str.length();
-    int count = 0;
-    int offset = 0;
-
-    first = ff::stof(str, length, &count);
-    if (count == 0) {
-        // No numbers found, return zero extrusion
-        return Extrude(0, 0);
+    if (node.IsSequence() && node.size() >= 2) {
+        if (YamlUtil::getFloat(node[0], first) && YamlUtil::getFloat(node[1], second)) {
+            // Got two numbers, so return an extrusion from first to second
+            return Extrude(first, second);
+        }
     }
 
-    // Advance and skip the delimiter character
-    offset += count;
-    if (length - offset <= 0) {
-        return Extrude(0, first);
-    }
-    while (str[offset] == ' ') { offset++; }
-    if (str[offset++] != ',') {
-        return Extrude(0, first);
-    }
-
-    // Get a float if possible & advance pos to the end of the number
-    second = ff::stof(str + offset, length - offset, &count);
-    if (count == 0) {
+    if (YamlUtil::getFloat(node, first)) {
```

Comment:
Will be simpler to just check if `first` is > 0, and extrude from 0 to `first`.

Reply:
Don't understand this comment - can you be more specific?

Reply:
```
if (first > 0) {
    return Extrude(0, first);
}
```

Instead of getting `float` value from the yaml node again, already have the `first` value parsed from the yaml node above on line 23.

Reply:
The block before this is for cases where `node` is a sequence, but this block is for scalars. They have to be accessed differently.

Reply:
Ohh yeah, missed that. All good.